rollup/core/classes/Rollup.cls

```diff
@@ -21,32 +19,18 @@ global without sharing virtual class Rollup implements Database.Batchable<SObjec
   private static RollupControl__mdt specificControl;
   @testVisible
   private static Integer maxQueryRowOverride;
-  // in this case, we promote the visiblity of this variable for tests
-  // specially to validate that the buffer is flushed when "processStoredFlowRollups" is called
   @testVisible
-  private static List<Rollup> CACHED_ROLLUPS = new List<Rollup>();
-  private static Boolean isRunningAsync = false;
-  private static Integer SENTINEL_COUNT_VALUE = -1;
-  private static Map<SObjectType, Set<TriggerOperation>> CACHED_APEX_OPERATIONS = new Map<SObjectType, Set<TriggerOperation>>();
+  private static List<RollupAsyncProcessor> CACHED_ROLLUPS = new List<RollupAsyncProcessor>();
```

Comment:
Yeah that’s a good point. I’ll add an interface for this!

Reply:
I realized after playing with the interface a bit that it would be a breaking change for anyone currently using the global static methods exposed off of `Rollup`. I was able to consolidate most of the properties into `RollupAsyncProcessor`, but there's still quite a bit of surface area inside of `Rollup` that ends up knowing about the type (also, take a look at the updated line 1959 in `Rollup` - that cast is evil, so I'll keep playing around with things to try to tighten everything up). Even in an ideal world, `Rollup` would still need to at least know about the existence of the class in order to call the `RollupAsyncProcessor.getProcessor` static factory method.